src/api/recommend_api.rs

```diff
@@ -2,24 +2,31 @@ use crate::common::helpers::process_response;
 use actix_web::rt::time::Instant;
 use actix_web::{post, web, Responder};
 use collection::operations::types::RecommendRequest;
+use segment::types::ScoredPoint;
 use std::sync::Arc;
+use storage::content_manager::errors::StorageError;
 use storage::content_manager::toc::TableOfContent;
 
+async fn do_recommend_points(
```

Comment:
Here and below had to move it out to it's own method as I can't call async method inside ``and_then`` - it accepts sync closure. But I would need it to be a separate method to be called from the GRPC anyway. I'll move them out into ``common`` module and remove do_ in the next PR where I expose actix as a feature and add grpc